File: php/features.php
<?php
declare(strict_types=1);

// RuntimebuzzArticle SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RuntimebuzzArticleFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RuntimebuzzArticleBaseFeature();
            case "test":
                return new RuntimebuzzArticleTestFeature();
            default:
                return new RuntimebuzzArticleBaseFeature();
        }
    }
}

File: py/runtimebuzzarticle_sdk/features.py
# RuntimebuzzArticle SDK feature factory

from runtimebuzzarticle_sdk.feature.base_feature import RuntimebuzzArticleBaseFeature
from runtimebuzzarticle_sdk.feature.test_feature import RuntimebuzzArticleTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RuntimebuzzArticleBaseFeature(),
        "test": lambda: RuntimebuzzArticleTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
